### PR TITLE
[Merged by Bors] - feat: Topology.Constructions. add isOpen_pi_iff

### PR DESCRIPTION
**Pull request by xroblot**
This is the mathlib4 pendant of https://github.com/leanprover-community/mathlib/pull/18390

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
LGTM

bors d+

Please merge roughly at the same time as the mathlib3 PR

**Comment by eric-wieser** [comment]
> Please merge roughly at the same time as the mathlib3 PR

Please don't do this. Instead:
* wait for the mathlib3 PR to be merged
* *wait for the mathport output
* replace the manual port with mathport output
* update the commit sha

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Patrick Massot
 
 ! This file was ported from Lean 3 source module topology.constructions
-! leanprover-community/mathlib commit bcfa726826abd57587355b4b5b7e78ad6527b7e4
+! leanprover-community/mathlib commit dc6c365e751e34d100e80fe6e314c3c3e0fd2988
```
Comment (eric-wieser) [inline Mathlib/Topology/Constructions.lean:7]: I think this conflicts with #2176

Comment (eric-wieser) [reply Mathlib/Topology/Constructions.lean:7]: Namely, this incorrectly marks this file as up to date, but did not forward-port all the necessary changes

Comment (xroblot) [reply Mathlib/Topology/Constructions.lean:7]: Oh sorry, what should I do then?

Comment (eric-wieser) [reply Mathlib/Topology/Constructions.lean:7]: In future, always check the port status page for a forward-port; if there is another PR that changes the file before yours, then mark your PR as dependent on that one.

I intend to build a bot that pastes these links automatically

**Pull request closed without merging**